### PR TITLE
Follow Unfollow option

### PR DESCRIPTION
**Pull request by shakil-fg**

### CODE REVIEW

**Comment by mdmahmudurrahman** [comment]
OK

File: app/views/users/_follow.html.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<%= form_for(current_user.active_relationships.build, remote: true) do |f| %>
```
Comment (MuhammadTamzid) [inline app/views/users/_follow.html.erb]: remove ()

File: app/views/users/_unfollow.html.erb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<%= form_for(current_user.active_relationships.find_by(followed_id: @user.id),
```
Comment (MuhammadTamzid) [inline app/views/users/_unfollow.html.erb]: remove ()

**Comment by MuhammadTamzid** [comment]
ok

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -22,6 +22,18 @@ def update
     end   
   end
 
+  def following
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: pls follow RESTFUL

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -22,6 +22,18 @@ def update
...
+    @title = t :following
+    @users = @user.following
+    render "show_follow"
+  end
+
+  def followers
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: pls follow RESTFUL

File: app/models/relationship.rb [hunk 5]
```diff
@@ -1,4 +1,6 @@
 class Relationship < ActiveRecord::Base
   belongs_to :follower, class_name: User.name
   belongs_to :followed, class_name: User.name
+  validates :follower, presence: true
```
Comment (dieunb) [inline app/models/relationship.rb]: split to many block code

**Pull request merged**